### ISSUE TITLE
v6.4.0 OnConnect hook bug?

### ISSUE THREAD

**Issue by xkeyideal** [go-redis/redis#572]
see #565 
I test use v6.4.0 by OnConnect hook to exec Client SetName command, but the bug of  error " redis: pipeline is empty"
The source code:

```
cb := func(cn *redis.Conn) error {
		return cn.ClientSetName("client setname seconds").Err()
	}
	client := redis.NewClusterClient(&redis.ClusterOptions{
		Addrs:              []string{"10.100.150.24:6316", "10.100.150.31:6199", "10.100.188.79:6221"},
		Password:           "",
		DialTimeout:        time.Duration(1000) * time.Millisecond,
		WriteTimeout:       time.Duration(1000) * time.Millisecond,
		ReadTimeout:        time.Duration(1000) * time.Millisecond,
		PoolSize:           10,
		PoolTimeout:        time.Duration(30) * time.Second,
		IdleTimeout:        time.Duration(300) * time.Millisecond,
		IdleCheckFrequency: time.Duration(120) * time.Second,
		OnConnect:          cb,
	})

	fmt.Println(client.ClientList().Result())
	client.Close()
```
![image](https://cloud.githubusercontent.com/assets/6369390/26567242/41083efa-452c-11e7-91b9-01336e762f39.png)

I not use OnConnect hook, source code:
```
//	cb := func(cn *redis.Conn) error {
	//		return cn.ClientSetName("client setname seconds").Err()
	//	}
	client := redis.NewClusterClient(&redis.ClusterOptions{
		Addrs:              []string{"10.100.150.24:6316", "10.100.150.31:6199", "10.100.188.79:6221"},
		Password:           "",
		DialTimeout:        time.Duration(1000) * time.Millisecond,
		WriteTimeout:       time.Duration(1000) * time.Millisecond,
		ReadTimeout:        time.Duration(1000) * time.Millisecond,
		PoolSize:           10,
		PoolTimeout:        time.Duration(30) * time.Second,
		IdleTimeout:        time.Duration(300) * time.Millisecond,
		IdleCheckFrequency: time.Duration(120) * time.Second,
		//	OnConnect:          cb,
	})

	fmt.Println(client.ClientList().Result())
	client.Close()
```
The result is ok (the last redis client disconnect):
![image](https://cloud.githubusercontent.com/assets/6369390/26567257/610c45ca-452c-11e7-8c12-006ea7bd7454.png)

**Comment by xkeyideal** [comment]
If set OnConnect is not nil, the redis.go file initConn function conn.Piplelined will return err.
"redis: pipeline is empty"

**Comment by vmihailenco** [comment]
Should be fixed in v6.4.1
